src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -32,6 +31,10 @@ public static int IndexOf<T>(this Span<T> span, T value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int IndexOf(this Span<byte> span, byte value)
         {
+            if (!BitConverter.IsLittleEndian)
```

Comment:
Ah, makes sense. Do you know how much this branch add to "early" searches? i.e. search that finds the byte at index 0.